test_runner/random_test_runner/src/lanelet_utils.cpp

```diff
@@ -0,0 +1,292 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Developed by Robotec.ai.
+
+#include "random_test_runner/lanelet_utils.hpp"
+
+#include <lanelet2_core/LaneletMap.h>
+#include <lanelet2_core/geometry/Lanelet.h>
+#include <lanelet2_core/primitives/BasicRegulatoryElements.h>
+#include <lanelet2_core/primitives/LaneletSequence.h>
+#include <lanelet2_io/Io.h>
+#include <lanelet2_routing/Route.h>
+#include <lanelet2_routing/RoutingCost.h>
+#include <lanelet2_traffic_rules/TrafficRulesFactory.h>
+
+#include <lanelet2_extension_psim/projection/mgrs_projector.hpp>
+
+#include "geographic_msgs/msg/geo_point.hpp"
+#include "traffic_simulator/hdmap_utils/hdmap_utils.hpp"
+#include "traffic_simulator/math/linear_algebra.hpp"
+
+LaneletUtils::LaneletUtils(const boost::filesystem::path & filename)
+{
+  lanelet::projection::MGRSProjector projector;
+  lanelet::ErrorMessages errors;
+  lanelet_map_ptr_ = lanelet::load(filename.string(), projector, &errors);
+
+  double laneChangeCost = 2;
+  lanelet::routing::RoutingCostPtrs costPtrs{
+    std::make_shared<lanelet::routing::RoutingCostDistance>(laneChangeCost)};
+  auto traffic_rules_vehicle_ptr = lanelet::traffic_rules::TrafficRulesFactory::create(
+    lanelet::Locations::Germany, lanelet::Participants::Vehicle);
+  vehicle_routing_graph_ptr_ =
+    lanelet::routing::RoutingGraph::build(*lanelet_map_ptr_, *traffic_rules_vehicle_ptr, costPtrs);
+
+  hdmap_utils_ptr_ =
+    std::make_shared<hdmap_utils::HdMapUtils>(filename, geographic_msgs::msg::GeoPoint());
+}
+
+std::vector<int64_t> LaneletUtils::getLaneletIds() { return hdmap_utils_ptr_->getLaneletIds(); }
+
+geometry_msgs::msg::PoseStamped LaneletUtils::toMapPose(
+  traffic_simulator_msgs::msg::LaneletPose lanelet_pose)
+{
+  return hdmap_utils_ptr_->toMapPose(lanelet_pose);
+}
+
+std::vector<std::int64_t> LaneletUtils::getRoute(
+  std::int64_t from_lanelet_id, std::int64_t to_lanelet_id)
+{
+  return hdmap_utils_ptr_->getRoute(from_lanelet_id, to_lanelet_id);
+}
+
+double LaneletUtils::getLaneletLength(std::int64_t lanelet_id)
+{
+  return hdmap_utils_ptr_->getLaneletLength(lanelet_id);
+}
+
+double LaneletUtils::computeDistance(
+  traffic_simulator_msgs::msg::LaneletPose p1, traffic_simulator_msgs::msg::LaneletPose p2)
+{
+  auto p1_g = hdmap_utils_ptr_->toMapPose(p1).pose.position;
+  auto p2_g = hdmap_utils_ptr_->toMapPose(p2).pose.position;
+  geometry_msgs::msg::Point d;
+  d.x = p1_g.x - p2_g.x;
+  d.y = p1_g.y - p2_g.y;
+  d.z = p1_g.z - p2_g.z;
+  return std::sqrt(d.x * d.x + d.y * d.y + d.z * d.z);
```

Comment:
`std::hypot`?

Reply:
Turns out three-argument `std::hypot` is not supported in c++14
https://en.cppreference.com/w/cpp/numeric/math/hypot